main.ts

```diff
@@ -1,137 +1,85 @@
-					return true;
-				}
-			}
-		});
-
-		// This adds a settings tab so the user can configure various aspects of the plugin
-		this.addSettingTab(new SampleSettingTab(this.app, this));
-
-		// If the plugin hooks up any global DOM events (on parts of the app that doesn't belong to this plugin)
-		// Using this function will automatically remove the event listener when this plugin is disabled.
-		this.registerDomEvent(document, 'click', (evt: MouseEvent) => {
-			console.log('click', evt);
-		});
-
-		// When registering intervals, this function will automatically clear the interval when the plugin is disabled.
-		this.registerInterval(window.setInterval(() => console.log('setInterval'), 5 * 60 * 1000));
-	}
-
-	onunload() {
-
-	}
-
-	async loadSettings() {
-		this.settings = Object.assign({}, DEFAULT_SETTINGS, await this.loadData());
-	}
-
-	async saveSettings() {
-		await this.saveData(this.settings);
-	}
-}
-
-class SampleModal extends Modal {
-	constructor(app: App) {
-		super(app);
-	}
-
-	onOpen() {
-		const {contentEl} = this;
-		contentEl.setText('Woah!');
-	}
-
-	onClose() {
-		const {contentEl} = this;
-		contentEl.empty();
-	}
-}
-
-class SampleSettingTab extends PluginSettingTab {
-	plugin: MyPlugin;
-
-	constructor(app: App, plugin: MyPlugin) {
-		super(app, plugin);
-		this.plugin = plugin;
-	}
-
-	display(): void {
-		const {containerEl} = this;
-
-		containerEl.empty();
-
-		containerEl.createEl('h2', {text: 'Settings for my awesome plugin.'});
-
-		new Setting(containerEl)
-			.setName('Setting #1')
-			.setDesc('It\'s a secret')
-			.addText(text => text
-				.setPlaceholder('Enter your secret')
-				.setValue(this.plugin.settings.mySetting)
-				.onChange(async (value) => {
-					console.log('Secret: ' + value);
-					this.plugin.settings.mySetting = value;
-					await this.plugin.saveSettings();
-				}));
-	}
+import { Plugin, MarkdownPostProcessorContext } from 'obsidian';
+
+const toFixed = (source: string, significantDigits: number): string => (
+  Number.parseFloat(source).toFixed(significantDigits)
+);
+
+type BjcpStat = {
+  low?: string;
+  high?: string;
+  flexible?: boolean;
+};
+const renderStatRange = (
+  stat: BjcpStat,
+  significantDigits: number = 0,
+  suffix: string = ''
+): string => {
+  if (stat.flexible === true) {
+    return '<em>Varies</em>';
+  } else {
+    return `${toFixed(stat.low, significantDigits)}-${toFixed(stat.high, significantDigits)}${suffix}`;
+  }
+};
+
+export default class BjcpPlugin extends Plugin {
+  async onload() {
+    this.registerMarkdownPostProcessor((el: HTMLElement, ctx: MarkdownPostProcessorContext) => {
+      console.log('el => ', el);
+      console.log('ctx => ', ctx);
+      const { frontmatter } = ctx;
+
+      if (!frontmatter.style || !frontmatter.style_name || !frontmatter.stats) {
+        return;
+      }
+
+      let frontMatterContainer = el.querySelector('.frontmatter-container');

```

Comment:
Unless I'm missing something in the API, there isn't really a good way to tell _where_ on the page you are rendering. This solution is a hack because I'm looking for the container that holds the aliases and tags from the front matter -- if I see the element with that class then I can _reasonably_ assume that I'm in the first section that's being rendered.